ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
+
+    TcpHandler<II, OO> intercept(TcpHandler<I, O> handler);
```

Comment:
I sketched out providing for interceptors on the connection by separating the reading and writing. This allows for different interceptors (and even multiple interceptors) on the read and write which I think will be necessary since reading and writing on a connection is often unbalanced (e.g. more emphasis is often placed on reading data and figuring out what to do than on what's written out to the client, since it can often be nothing more than an ACK).

https://gist.github.com/jbrisbin/686e565660b27b9b0f91